osp/core/ontology/namespace.py

```diff
@@ -113,86 +130,233 @@ def __getitem__(self, label):
 
         Args:
-            name(str): The name of the ontology entity
-            _case_sensitive(bool): Name should be case sensitive,
-                defaults to False
-            _force_by_iri(bool):  Name is IRI suffix,
-                defaults to False
+            iri (Union[str, rdlib.URIRef]): The iri of the ontology entity.
+            _name (str): Not mean to be provided by the user. Just passed to
+                        the `from_iri` method of the namespace registry.
 
         Returns:
-            OntologyEntity: The ontology entity
+            OntologyEntity: The ontology entity.
 
+        Raises:
+            KeyError: When the iri does not belong to the namespace.
         """
+        if rdflib.URIRef(str(iri)) in self:
+            return self._namespace_registry.from_iri(str(iri), _name=_name)
+        else:
+            raise KeyError(f"The IRI {iri} does not belong to the namespace"
+                           f"{self}.")
+
+    def get_from_suffix(self, suffix, case_sensitive=False):
+        """Get an ontology entity from its namespace suffix.
+
+        Args:
+            suffix (str): Suffix of the ontology entity.
+            case_sensitive (bool): Whether to search also for the same suffix
+                                   with different capitalization. By default,
+                                   such a search is performed.
+        """
+        iri = rdflib.URIRef(str(self._iri) + suffix)
         try:
-            return self._do_get(name, _case_sensitive, _force_by_iri)
+            return self.get_from_iri(iri, _name=suffix)
         except KeyError as e:
-            if name and name.startswith("INVERSE_OF_"):
-                return self._do_get(name[11:], _case_sensitive,
-                                    _force_by_iri).inverse
+            if not case_sensitive:
+                return self._get_case_insensitive(suffix,
+                                                  self.get_from_suffix)
             raise e
 
-    def _do_get(self, name, _case_sensitive, _force_by_iri):
+    def _get_from_label(self, label, lang=None, case_sensitive=False):
+        """Get an ontology entity from the registry by label.
+
+        Args:
+            label (str): The label of the ontology entity.
+
+        Raises:
+            KeyError: Unknown label.
+
+        Returns:
+            OntologyEntity: The ontology entity.
+        """
+        results = []
+        for iri in self._get_namespace_subjects():
+            entity_labels = self._get_labels_for_iri(iri, lang=lang)
+            if case_sensitive is False:
+                entity_labels = (label.lower() for label in entity_labels)
+                comp_label = label.lower()
+            else:
+                comp_label = label
+            if comp_label in entity_labels:
+                _name = label if self._reference_by_label else None
+                results.append(self.get_from_iri(iri, _name=_name))
+        else:
```

Comment:
I'm a bit confused about these for-else stamements that you regulary use.
In this case there is no break in the loop, so couldn't you just omit the else?

Reply:
Yes, in this case it does not make a difference. I'll omit it in a commit that I am pushing in a while.